src/lib/LED/LED.h

```diff
@@ -23,104 +23,144 @@ void WS281BsetLED(uint32_t color)
     strip.ClearTo(RgbColor(HtmlColor(color)));
     strip.Show();
 }
+#endif
+
+#if WS2812_LED_IS_USED || (defined(PLATFORM_ESP32) && defined(GPIO_PIN_LED))
+static enum {
+    STARTUP = 0,
+    NORMAL = 1
+} state;
+static unsigned long blinkyUpdateTime;
+static uint8_t hue = 0, saturation = 255, lightness = 128;
+static uint8_t hueStepValue = 1;
+static uint8_t lightnessStep = 5;
 
-void WS281BsetLED(uint8_t const r, uint8_t const g, uint8_t const b) // takes RGB data
+static uint32_t HsvToRgb(uint8_t hue, uint8_t saturation, uint8_t lightness)
 {
-    strip.ClearTo(RgbColor(r, g, b));
-    strip.Show();
+    uint8_t region, remainder, p, q, t;
+
+    if (saturation == 0)
+    {
+        return lightness << 16 | lightness << 8 | lightness;
+    }
+
+    region = hue / 43;
+    remainder = (hue - (region * 43)) * 6; 
+
+    p = (lightness * (255 - saturation)) >> 8;
+    q = (lightness * (255 - ((saturation * remainder) >> 8))) >> 8;
+    t = (lightness * (255 - ((saturation * (255 - remainder)) >> 8))) >> 8;
+
+    switch (region)
+    {
+        case 0:
+            return lightness << 16 | t << 8 | p;
+        case 1:
+            return q << 16 | lightness << 8 | p;
+        case 2:
+            return p << 16 | lightness << 8 | t;
+        case 3:
+            return p << 16 | q << 8 | lightness;
+        case 4:
+            return t << 16 | p << 8 | lightness;
+        default:
+            return lightness << 16 | p << 8 | q;
+    }
+}
+
+static void blinkyUpdate() {
+    unsigned long now = millis();
```

Comment:
Pass now from updateLEDs and remove the millis() call?

Reply:
Good call